Tests not running anymore on my Ubuntu 22.04

```
$ mc
The Meson build system
Version: 0.61.2
Source dir: /home/kelson/code/libkiwix
Build dir: /home/kelson/code/libkiwix/build
Build type: native build
Project name: libkiwix
Project version: 13.0.0
C++ compiler for the host machine: c++ (gcc 11.4.0 "c++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C++ linker for the host machine: c++ ld.bfd 2.38
Host machine cpu family: x86_64
Host machine cpu: x86_64
Run-time dependency threads found: YES
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Run-time dependency icu-i18n found: YES 70.1
Run-time dependency pugixml found: YES 1.12
Run-time dependency libcurl found: YES 7.81.0
Run-time dependency libmicrohttpd found: YES 0.9.75
Run-time dependency zlib found: YES 1.2.11
Run-time dependency xapian-core found: YES 1.4.18
Has header "mustache.hpp" : NO 
Has header "mustache.hpp" : YES 
Run-time dependency libzim found: YES 9.1.0
Header <zim/zim.h> has symbol "LIBZIM_WITH_XAPIAN" with dependency libzim: YES 
Program kiwix-resources found: YES (/home/kelson/code/libkiwix/scripts/kiwix-resources)
Program kiwix-compile-resources found: YES (/home/kelson/code/libkiwix/scripts/kiwix-compile-resources)
Program kiwix-compile-i18n found: YES (/home/kelson/code/libkiwix/scripts/kiwix-compile-i18n)
Program python3 found: YES (/usr/bin/python3)
Configuring kiwix_config.h using configuration
Run-time dependency GTest found: YES (building self)
Program python3 found: YES (/usr/bin/python3)
Configuring example.zim with command
Program python3 found: YES (/usr/bin/python3)
Configuring zimfile.zim with command
Program python3 found: YES (/usr/bin/python3)
Configuring zimfile&other.zim with command
Program python3 found: YES (/usr/bin/python3)
Configuring corner_cases#&.zim with command
Program python3 found: YES (/usr/bin/python3)
Configuring poor.zim with command
Program python3 found: YES (/usr/bin/python3)
Configuring library.xml with command
Program python3 found: YES (/usr/bin/python3)
Configuring lib_for_server_search_test.xml with command
Program python3 found: YES (/usr/bin/python3)
Configuring customized_resources.txt with command
Program python3 found: YES (/usr/bin/python3)
Configuring helloworld.txt with command
Program python3 found: YES (/usr/bin/python3)
Configuring welcome.html with command
Configuring kiwix.pc using configuration
Build targets in project: 24

Found ninja-1.10.1 at /usr/bin/ninja
ninja: Entering directory `build'
[1/121] Generating static/preprocessed_resource_files with a custom command
Preprocessing skin/caret.png ...
Preprocessing skin/bittorrent.png ...
Preprocessing skin/magnet.png ...
Preprocessing skin/feed.svg ...
Preprocessing skin/langSelector.svg ...
Preprocessing skin/download.png ...
Preprocessing skin/hash.png ...
Preprocessing skin/search-icon.svg ...
Preprocessing skin/iso6391To3.js ...
Preprocessing skin/isotope.pkgd.min.js ...
Preprocessing skin/index.js ...
Preprocessing skin/autoComplete/autoComplete.min.js ...
Preprocessing skin/kiwix.css ...
Preprocessing skin/fonts/Poppins.ttf ...
Preprocessing skin/fonts/Roboto.ttf ...
Preprocessing skin/taskbar.css ...
Preprocessing skin/index.css ...
Preprocessing skin/search_results.css ...
Preprocessing skin/blank.html ...
Preprocessing skin/polyfills.js ...
Preprocessing skin/viewer.js ...
Preprocessing skin/i18n.js ...
Preprocessing skin/mustache.min.js ...
Preprocessing skin/languages.js ...
Preprocessing viewer.html ...
Preprocessing skin/autoComplete/css/autoComplete.css ...
Preprocessing templates/search_result.html ...
Preprocessing templates/search_result.xml ...
Preprocessing templates/error.html ...
Preprocessing templates/error.xml ...
Preprocessing templates/index.html ...
Preprocessing skin/favicon/apple-touch-icon.png ...
Preprocessing skin/favicon/favicon-32x32.png ...
Preprocessing skin/favicon/favicon-16x16.png ...
Preprocessing skin/favicon/site.webmanifest ...
Preprocessing skin/favicon/safari-pinned-tab.svg ...
Preprocessing skin/favicon/favicon.ico ...
Preprocessing skin/favicon/browserconfig.xml ...
Preprocessing templates/suggestion.json ...
Preprocessing templates/captured_external.html ...
Preprocessing templates/catalog_entries.xml ...
Preprocessing templates/catalog_v2_root.xml ...
Preprocessing templates/catalog_v2_entries.xml ...
Preprocessing templates/catalog_v2_entry.xml ...
Preprocessing templates/catalog_v2_partial_entry.xml ...
Preprocessing templates/catalog_v2_categories.xml ...
Preprocessing templates/catalog_v2_languages.xml ...
Preprocessing templates/url_of_search_results_css ...
Preprocessing templates/viewer_settings.js ...
Preprocessing templates/no_js_library_page.html ...
Preprocessing templates/no_js_download.html ...
Preprocessing opensearchdescription.xml ...
Preprocessing ft_opensearchdescription.xml ...
Preprocessing catalog_v2_searchdescription.xml ...
Preprocessing skin/favicon/android-chrome-192x192.png ...
Preprocessing skin/favicon/android-chrome-512x512.png ...
Preprocessing skin/favicon/mstile-70x70.png ...
Preprocessing skin/favicon/mstile-144x144.png ...
Preprocessing skin/favicon/mstile-150x150.png ...
Preprocessing skin/favicon/mstile-310x150.png ...
Preprocessing skin/favicon/mstile-310x310.png ...
[2/121] Generating static/i18n_resources with a custom command
ar.json -> ar
bn.json -> bn
br.json -> br
cs.json -> cs
dag.json -> dag
de.json -> de
dga.json -> dga
el.json -> el
en.json -> en
es.json -> es
fi.json -> fi
fr.json -> fr
ha.json -> ha
he.json -> he
hi.json -> hi
hy.json -> hy
ia.json -> ia
ig.json -> ig
it.json -> it
ja.json -> ja
ko.json -> ko
ku-latn.json -> ku_latn
lb.json -> lb
mk.json -> mk
ms.json -> ms
nl.json -> nl
nqo.json -> nqo
or.json -> or
pl.json -> pl
ru.json -> ru
sc.json -> sc
sk.json -> sk
skr-arab.json -> skr_arab
sl.json -> sl
sq.json -> sq
sv.json -> sv
te.json -> te
test.json -> test
tr.json -> tr
zh-hans.json -> zh_hans
zh-hant.json -> zh_hant
[3/121] Generating static/resources with a custom command
skin/caret.png ['skin', 'caret_png']
skin/bittorrent.png ['skin', 'bittorrent_png']
skin/magnet.png ['skin', 'magnet_png']
skin/feed.svg ['skin', 'feed_svg']
skin/langSelector.svg ['skin', 'langSelector_svg']
skin/download.png ['skin', 'download_png']
skin/hash.png ['skin', 'hash_png']
skin/search-icon.svg ['skin', 'search_icon_svg']
skin/iso6391To3.js ['skin', 'iso6391To3_js']
skin/isotope.pkgd.min.js ['skin', 'isotope_pkgd_min_js']
skin/index.js ['skin', 'index_js']
skin/autoComplete/autoComplete.min.js ['skin', 'autoComplete', 'autoComplete_min_js']
skin/kiwix.css ['skin', 'kiwix_css']
skin/taskbar.css ['skin', 'taskbar_css']
skin/index.css ['skin', 'index_css']
skin/fonts/Poppins.ttf ['skin', 'fonts', 'Poppins_ttf']
skin/fonts/Roboto.ttf ['skin', 'fonts', 'Roboto_ttf']
skin/search_results.css ['skin', 'search_results_css']
skin/blank.html ['skin', 'blank_html']
skin/polyfills.js ['skin', 'polyfills_js']
skin/viewer.js ['skin', 'viewer_js']
skin/i18n.js ['skin', 'i18n_js']
skin/languages.js ['skin', 'languages_js']
skin/mustache.min.js ['skin', 'mustache_min_js']
viewer.html ['viewer_html']
templates/search_result.html ['templates', 'search_result_html']
templates/search_result.xml ['templates', 'search_result_xml']
templates/error.html ['templates', 'error_html']
templates/error.xml ['templates', 'error_xml']
templates/index.html ['templates', 'index_html']
templates/suggestion.json ['templates', 'suggestion_json']
templates/captured_external.html ['templates', 'captured_external_html']
templates/catalog_entries.xml ['templates', 'catalog_entries_xml']
templates/catalog_v2_root.xml ['templates', 'catalog_v2_root_xml']
templates/catalog_v2_entries.xml ['templates', 'catalog_v2_entries_xml']
templates/catalog_v2_entry.xml ['templates', 'catalog_v2_entry_xml']
templates/catalog_v2_partial_entry.xml ['templates', 'catalog_v2_partial_entry_xml']
templates/catalog_v2_categories.xml ['templates', 'catalog_v2_categories_xml']
templates/catalog_v2_languages.xml ['templates', 'catalog_v2_languages_xml']
templates/url_of_search_results_css ['templates', 'url_of_search_results_css']
templates/viewer_settings.js ['templates', 'viewer_settings_js']
templates/no_js_library_page.html ['templates', 'no_js_library_page_html']
templates/no_js_download.html ['templates', 'no_js_download_html']
opensearchdescription.xml ['opensearchdescription_xml']
ft_opensearchdescription.xml ['ft_opensearchdescription_xml']
catalog_v2_searchdescription.xml ['catalog_v2_searchdescription_xml']
skin/autoComplete/css/autoComplete.css ['skin', 'autoComplete', 'css', 'autoComplete_css']
skin/favicon/android-chrome-192x192.png ['skin', 'favicon', 'android_chrome_192x192_png']
skin/favicon/android-chrome-512x512.png ['skin', 'favicon', 'android_chrome_512x512_png']
skin/favicon/apple-touch-icon.png ['skin', 'favicon', 'apple_touch_icon_png']
skin/favicon/browserconfig.xml ['skin', 'favicon', 'browserconfig_xml']
skin/favicon/favicon-16x16.png ['skin', 'favicon', 'favicon_16x16_png']
skin/favicon/favicon-32x32.png ['skin', 'favicon', 'favicon_32x32_png']
skin/favicon/favicon.ico ['skin', 'favicon', 'favicon_ico']
skin/favicon/mstile-70x70.png ['skin', 'favicon', 'mstile_70x70_png']
skin/favicon/mstile-144x144.png ['skin', 'favicon', 'mstile_144x144_png']
skin/favicon/mstile-150x150.png ['skin', 'favicon', 'mstile_150x150_png']
skin/favicon/mstile-310x150.png ['skin', 'favicon', 'mstile_310x150_png']
skin/favicon/mstile-310x310.png ['skin', 'favicon', 'mstile_310x310_png']
skin/favicon/safari-pinned-tab.svg ['skin', 'favicon', 'safari_pinned_tab_svg']
skin/favicon/site.webmanifest ['skin', 'favicon', 'site_webmanifest']
skin/i18n/ar.json ['skin', 'i18n', 'ar_json']
skin/i18n/bn.json ['skin', 'i18n', 'bn_json']
skin/i18n/br.json ['skin', 'i18n', 'br_json']
skin/i18n/cs.json ['skin', 'i18n', 'cs_json']
skin/i18n/dag.json ['skin', 'i18n', 'dag_json']
skin/i18n/de.json ['skin', 'i18n', 'de_json']
skin/i18n/dga.json ['skin', 'i18n', 'dga_json']
skin/i18n/el.json ['skin', 'i18n', 'el_json']
skin/i18n/en.json ['skin', 'i18n', 'en_json']
skin/i18n/es.json ['skin', 'i18n', 'es_json']
skin/i18n/fi.json ['skin', 'i18n', 'fi_json']
skin/i18n/fr.json ['skin', 'i18n', 'fr_json']
skin/i18n/ha.json ['skin', 'i18n', 'ha_json']
skin/i18n/he.json ['skin', 'i18n', 'he_json']
skin/i18n/hi.json ['skin', 'i18n', 'hi_json']
skin/i18n/hy.json ['skin', 'i18n', 'hy_json']
skin/i18n/ia.json ['skin', 'i18n', 'ia_json']
skin/i18n/ig.json ['skin', 'i18n', 'ig_json']
skin/i18n/it.json ['skin', 'i18n', 'it_json']
skin/i18n/ja.json ['skin', 'i18n', 'ja_json']
skin/i18n/ko.json ['skin', 'i18n', 'ko_json']
skin/i18n/ku-latn.json ['skin', 'i18n', 'ku_latn_json']
skin/i18n/lb.json ['skin', 'i18n', 'lb_json']
skin/i18n/mk.json ['skin', 'i18n', 'mk_json']
skin/i18n/ms.json ['skin', 'i18n', 'ms_json']
skin/i18n/nl.json ['skin', 'i18n', 'nl_json']
skin/i18n/nqo.json ['skin', 'i18n', 'nqo_json']
skin/i18n/or.json ['skin', 'i18n', 'or_json']
skin/i18n/pl.json ['skin', 'i18n', 'pl_json']
skin/i18n/ru.json ['skin', 'i18n', 'ru_json']
skin/i18n/sc.json ['skin', 'i18n', 'sc_json']
skin/i18n/sk.json ['skin', 'i18n', 'sk_json']
skin/i18n/skr-arab.json ['skin', 'i18n', 'skr_arab_json']
skin/i18n/sl.json ['skin', 'i18n', 'sl_json']
skin/i18n/sq.json ['skin', 'i18n', 'sq_json']
skin/i18n/sv.json ['skin', 'i18n', 'sv_json']
skin/i18n/te.json ['skin', 'i18n', 'te_json']
skin/i18n/test.json ['skin', 'i18n', 'test_json']
skin/i18n/tr.json ['skin', 'i18n', 'tr_json']
skin/i18n/zh-hans.json ['skin', 'i18n', 'zh_hans_json']
skin/i18n/zh-hant.json ['skin', 'i18n', 'zh_hant_json']
[16/121] Compiling C++ object test/library.p/library.cpp.o
FAILED: test/library.p/library.cpp.o 
c++ -Itest/library.p -Iinclude -I../include -I/usr/include/kainjow -I/usr/local/include -I/usr/include/x86_64-linux-gnu -I/usr/include/p11-kit-1 -I/usr/src/googletest/googletest -I/usr/src/googletest/googletest/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++17 -g -pthread -MD -MQ test/library.p/library.cpp.o -MF test/library.p/library.cpp.o.d -o test/library.p/library.cpp.o -c ../test/library.cpp
In file included from ../test/library.cpp:20:
/usr/src/googletest/googletest/include/gtest/gtest.h: In instantiation of ‘testing::AssertionResult testing::internal::CmpHelperEQ(const char*, const char*, const T1&, const T2&) [with T1 = long unsigned int; T2 = int]’:
/usr/src/googletest/googletest/include/gtest/gtest.h:1564:23:   required from ‘static testing::AssertionResult testing::internal::EqHelper::Compare(const char*, const char*, const T1&, const T2&) [with T1 = long unsigned int; T2 = int; typename std::enable_if<((! std::is_integral<_Tp>::value) || (! std::is_pointer<_Dp>::value))>::type* <anonymous> = 0]’
../test/library.cpp:482:5:   required from here
/usr/src/googletest/googletest/include/gtest/gtest.h:1545:11: error: comparison of integer expressions of different signedness: ‘const long unsigned int’ and ‘const int’ [-Werror=sign-compare]
 1545 |   if (lhs == rhs) {
      |       ~~~~^~~~~~
cc1plus: all warnings being treated as errors
[25/121] Compiling C++ object src/libkiwix.so.13.0.0.p/library.cpp.o
ninja: build stopped: subcommand failed.
``

--- a/test/library.cpp
+++ b/test/library.cpp
@@ -479,11 +479,11 @@ TEST_F(LibraryTest, bookmarksSerializationTest)
     auto onlyValidBookmarks = new_lib->getBookmarks();
     auto allBookmarks = new_lib->getBookmarks(false);
 
-    ASSERT_EQ(onlyValidBookmarks.size(), 2);
+    ASSERT_EQ(onlyValidBookmarks.size(), 2U);
     EXPECT_EQ(onlyValidBookmarks[0].getBookId(), bookId1);
     EXPECT_EQ(onlyValidBookmarks[1].getBookId(), bookId2);
 
-    ASSERT_EQ(allBookmarks.size(), 3);
+    ASSERT_EQ(allBookmarks.size(), 3U);
     auto bookmark1 = allBookmarks[0];
     EXPECT_EQ(bookmark1.getBookId(), bookId1);
     EXPECT_EQ(bookmark1.getBookTitle(), book1.getTitle());
@@ -544,11 +544,11 @@ TEST_F(LibraryTest, MigrateBookmark)
     auto onlyValidBookmarks = lib->getBookmarks();
     auto allBookmarks = lib->getBookmarks(false);
 
-    ASSERT_EQ(onlyValidBookmarks.size(), 2);
+    ASSERT_EQ(onlyValidBookmarks.size(), 2U);
     EXPECT_EQ(onlyValidBookmarks[0].getBookId(), bookId1);
     EXPECT_EQ(onlyValidBookmarks[1].getBookId(), bookId2);
 
-    ASSERT_EQ(allBookmarks.size(), 6);
+    ASSERT_EQ(allBookmarks.size(), 6U);
     EXPECT_EQ(allBookmarks[0].getBookId(), bookId1);
     EXPECT_EQ(allBookmarks[1].getBookId(), "invalid-book-id");
     EXPECT_EQ(allBookmarks[2].getBookId(), bookId2);
@@ -563,14 +563,14 @@ TEST_F(LibraryTest, MigrateBookmark)
     onlyValidBookmarks = lib->getBookmarks();
     allBookmarks = lib->getBookmarks(false);
 
-    ASSERT_EQ(onlyValidBookmarks.size(), 5);
+    ASSERT_EQ(onlyValidBookmarks.size(), 5U);
     EXPECT_EQ(onlyValidBookmarks[0].getBookId(), bookId1);
     EXPECT_EQ(onlyValidBookmarks[1].getBookId(), bookId2);
     EXPECT_EQ(onlyValidBookmarks[2].getBookId(), bookId1+"_updated1yearlater");
     EXPECT_EQ(onlyValidBookmarks[3].getBookId(), bookId2);
     EXPECT_EQ(onlyValidBookmarks[4].getBookId(), bookId1+"_updated1yearlater_flavour");
 
-    ASSERT_EQ(allBookmarks.size(), 6);
+    ASSERT_EQ(allBookmarks.size(), 6U);
     EXPECT_EQ(allBookmarks[0].getBookId(), bookId1);
     EXPECT_EQ(allBookmarks[1].getBookId(), "invalid-book-id");
     EXPECT_EQ(allBookmarks[2].getBookId(), bookId2);
@@ -582,7 +582,7 @@ TEST_F(LibraryTest, MigrateBookmark)
 
     ASSERT_EQ(lib->migrateBookmarks(bookId1), 1);
     allBookmarks = lib->getBookmarks(false);
-    ASSERT_EQ(allBookmarks.size(), 6);
+    ASSERT_EQ(allBookmarks.size(), 6U);
     EXPECT_EQ(allBookmarks[0].getBookId(), bookId1+"_updated1yearlater");
     EXPECT_EQ(allBookmarks[1].getBookId(), "invalid-book-id");
     EXPECT_EQ(allBookmarks[2].getBookId(), bookId2);
@@ -596,14 +596,14 @@ TEST_F(LibraryTest, MigrateBookmark)
     onlyValidBookmarks = lib->getBookmarks();
     allBookmarks = lib->getBookmarks(false);
 
-    ASSERT_EQ(onlyValidBookmarks.size(), 5);
+    ASSERT_EQ(onlyValidBookmarks.size(), 5U);
     EXPECT_EQ(onlyValidBookmarks[0].getBookId(), bookId2);
     EXPECT_EQ(onlyValidBookmarks[1].getBookId(), bookId2);
     EXPECT_EQ(onlyValidBookmarks[2].getBookId(), bookId2);
     EXPECT_EQ(onlyValidBookmarks[3].getBookId(), bookId2);
     EXPECT_EQ(onlyValidBookmarks[4].getBookId(), bookId1+"_updated1yearlater_flavour");
 
-    ASSERT_EQ(allBookmarks.size(), 6);
+    ASSERT_EQ(allBookmarks.size(), 6U);
     EXPECT_EQ(allBookmarks[0].getBookId(), bookId2);
     EXPECT_EQ(allBookmarks[1].getBookId(), "invalid-book-id");
     EXPECT_EQ(allBookmarks[2].getBookId(), bookId2);
@@ -616,7 +616,7 @@ TEST_F(LibraryTest, MigrateBookmark)
     onlyValidBookmarks = lib->getBookmarks();
     allBookmarks = lib->getBookmarks(false);
 
-    ASSERT_EQ(onlyValidBookmarks.size(), 6);
+    ASSERT_EQ(onlyValidBookmarks.size(), 6U);
     EXPECT_EQ(onlyValidBookmarks[0].getBookId(), bookId2);
     EXPECT_EQ(onlyValidBookmarks[1].getBookId(), bookId1);
     EXPECT_EQ(onlyValidBookmarks[2].getBookId(), bookId2);
@@ -625,7 +625,7 @@ TEST_F(LibraryTest, MigrateBookmark)
     EXPECT_EQ(onlyValidBookmarks[5].getBookId(), bookId1+"_updated1yearlater_flavour");
 
 
-    ASSERT_EQ(allBookmarks.size(), 6);
+    ASSERT_EQ(allBookmarks.size(), 6U);
     EXPECT_EQ(allBookmarks[0].getBookId(), bookId2);
     EXPECT_EQ(allBookmarks[1].getBookId(), bookId1);
     EXPECT_EQ(allBookmarks[2].getBookId(), bookId2);

